### PR TITLE
DatePicker close button

### PR DESCRIPTION
**Pull request by kisiebel**
#### Pull request checklist

- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Adds an optional close button to the DatePicker. The button is located in the CalendarDay component. DatePicker, Calendar, and CalendarDay all take an optional prop showCloseButton that determines whether or not to render the button.

### CODE REVIEW

**Comment by kisiebel** [comment]
close button
Here's a screenshot!

The motivation for this button is accessibility - currently, on iOS with VoiceOver there is no way to close the DatePicker without selecting a date.

**Comment by betrue-final-final** [comment]
The weight of that glyph doesn't work very well with the up and down arrows. I would try another glyph. There's a few Xs.

**Comment by kisiebel** [comment]
Yeah that's true - I'll look for another one

**Comment by kisiebel** [comment]
image
Would this work better?

**Comment by betrue-final-final** [comment]
Much better!

**Comment by KevinTCoughlin** [comment]
Defer to @lorejoh12 as the component owner but it looks good and works for both `SPACE` and `ENTER`. I'd consider adding an example showing the close button to an existing modal example prior to merge.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -16,9 +16,11 @@ const styles: any = stylesImport;
 
 const leftArrow = 'Up';
 const rightArrow = 'Down';
+const closeIcon = 'CalculatorMultiply';
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: We use `Cancel` elsewhere.. not sure if that's the same glyph as this. This does fit-in with the UI if `Cancel` doesn't.

Comment (kisiebel) [reply packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Cancel is a bit too bold unfortunately, so I think this works better with the existing UI.

**Comment by kisiebel** [comment]
Thanks for the review! You're right that users can exit with ESC on most platforms, but on iOS there isn't really a good way to close the dialog with VoiceOver active.

**Pull request merged**